charts/castai-cluster-controller/templates/deployment.yaml

```
@@ -67,9 +68,20 @@ spec:
       nodeSelector:
         {{- toYaml . | nindent 8 }}
       {{- end }}
-      {{- with .Values.affinity }}
       affinity:
-        {{- toYaml . | nindent 8 }}
+        nodeAffinity:
+          {{- with .Values.nodeAffinity }}
+          {{- toYaml . | nindent 10 }}
+          {{- end }}
+        podAntiAffinity:
+          {{- with .Values.podAntiAffinity }}
+          {{- toYaml . | nindent 10 }}
+          {{- end }}
+      {{- if .Values.enableTopologySpreadConstraints }}
+      topologySpreadConstraints:
+       {{- with .Values.topologySpreadConstraints }}
+       {{- toYaml . | nindent 8 }}
+       {{- end }}
```

```suggestion
   {{- with .Values.topologySpreadConstraints }}
      topologySpreadConstraints:
       {{- toYaml . | nindent 8 }}
       {{- end }}
```

With this we will avoid empty topologySpreadConstraints key